src/clj/fluree/db/query/exec/eval.cljc

```diff
@@ -191,15 +191,17 @@
 
 (defn var->lang-var
   [var]
-  (-> var
-      (str "$-LANG")
-      symbol))
+  (when (where/variable? var)
```

Comment:
why is this check necessary? i don't think this should be called until we've already established that `var` is a variable, unless i'm missing something.

Reply:
This is required so that we can compare literals and nested expression results. In those cases we may (and do, there were broken tests before I added this) get numbers passed in, which results in a compile exception as we try to construct symbols like `101$-DATATYPE`.

If we're going to allow composing expressions arbitrarily we need this var-guard somewhere, and this seemed like the best place to put it.